charts/consul/templates/telemetry-collector-deployment.yaml

```diff
@@ -93,24 +93,31 @@ spec:
             valueFrom:
               fieldRef:
                 fieldPath: spec.nodeName
+          - name: CONSUL_NODE_NAME
+            value: $(NODE_NAME)-virtual
+          {{- include "consul.consulK8sConsulServerEnvVars" . | nindent 10 }}
+          {{- if .Values.global.enableConsulNamespaces }}
+          - name: CONSUL_NAMESPACE
+            value: {{ (default $defaults.consulNamespace .consulNamespace) }}
+          {{- end }}
           {{- if .Values.global.acls.manageSystemACLs }}
           - name: CONSUL_LOGIN_AUTH_METHOD
             value: {{ template "consul.fullname" . }}-k8s-auth-method
+          - name: CONSUL_LOGIN_DATACENTER
+            value: {{ .Values.global.datacenter }}
           - name: CONSUL_LOGIN_META
             value: "component=consul-telemetry-collector,pod=$(NAMESPACE)/$(POD_NAME)"
-          {{- end }}
-          - name: CONSUL_NODE_NAME
-            value: $(NODE_NAME)-virtual
-          {{- include "consul.consulK8sConsulServerEnvVars" . | nindent 10 }}
           {{- if .Values.global.enableConsulNamespaces }}
-          - name: CONSUL_NAMESPACE
-            value: {{ .Values.syncCatalog.consulNamespaces.consulDestinationNamespace }}
```

Comment:
this removes all ref to `.Values.syncCatalog.consulNamespaces.consulDestinationNamespace`. AFAICT, `.Values.syncCatalog.enabled` might be `false` but we'll still insert it here. My guess is that this is a bug that was from copying this setting from the Sync Catalog Deployment: https://github.com/hashicorp/consul-k8s/blob/768e29d2b7693e7900f75f7d0be1fd079d0c1244/charts/consul/templates/sync-catalog-deployment.yaml#L177